### DISCUSSION TITLE
helm chart release with multiple storageclass

### DISCUSSION THREAD

**Discussion by lzd-1230** [ceph/ceph-csi#4209] [Q&A] [answered]
Hi guys, now I got two ceph pool stand for a bounch of hdd and ssd respectfully.They are named "hdd-pool" and "ssd-pool".
I want to use ceph-csi-rbd installed by helm to provide storage for k8s, but i found that in `value.yaml` there‘s only one `StorageClass` fields for me to config. That means I shall make two `value.yaml` and install twice?
Can I config two  StorageClass in just one config file and make install once to create both of them? What config field do I need to focus to realize my target, thanks for advice!!

**Comment by Madhu-1** [comment, marked as answer]
@lzd-1230 i think its not a easy task to maintain the Storageclass today in helm templates as its hard to upgrade it and also we might remove support for creation of storageclass/snapshotclass from the helm and just ask the admin/user to create it externally using yaml files. 

>That means I shall make two value.yaml and install twice?

No, that's not going to helm and it can cause unwanted errors.

>Can I config two StorageClass in just one config file and make install once to create both of them? What config field do I need to focus to realize my target, thanks for advice!!

You can follow https://github.com/ceph/ceph-csi/blob/devel/examples/rbd/storageclass.yaml to create a new storageclass, in your case you just need to replace `pool` parameter in it.

**Reply by lzd-1230** [reply]
Thanks for your Prompt reply Madhu! As far as I successfully create `csi-rbd-provisioner` and `csi-rbd-nodeplugin`, then I can create storageClass by the separate file you mentioned, so I can create storageClass  the  as many as I can only if it all compare to a ceph-pool right?

**Reply by Madhu-1** [reply]
Yes you can create `N` number of storageclasses as per your requirements, please follow above storageclass yaml for reference
